tar/chunks/chunks_stats.c

```diff
@@ -414,3 +414,45 @@ chunks_stats_free(CHUNKS_S * C)
 	free(C->cachepath);
 	free(C);
 }
+
+/**
+ * chunks_initialize(const char * cachepath):
+ * Initialize the chunk directory (file) in the ${cachepath}.
+ */
+int
+chunks_initialize(const char * cachepath)
+{
+	struct chunks_stats_internal * C;
+	void * dir;
+
+	/* Allocate memory. */
+	if ((C = malloc(sizeof(struct chunks_stats_internal))) == NULL)
+		goto err0;
+
+	/* Create a copy of the path. */
+	if ((C->cachepath = strdup(cachepath)) == NULL)
+		goto err1;
+
+	/* Read directory (a useful way to initialize C). */
+	if ((C->HT = chunks_directory_read(cachepath, &dir,
+	    &C->stats_unique, &C->stats_total, &C->stats_extra, 0, 1)) == NULL)
+		goto err2;
+	C->dir = dir;
+
+	/* Write directory file to filesystem. */
+	if (chunks_fsck_end(C))
```

Comment:
1. because that function contains `chunks_directory_write()`, and frees the `CHUNK_S` cookie.  Despite the name, it doesn't look particularly "fsck-related" to me.

2. because I can't resist calling a function whose description contains "... , and close the fscking cookie".  :)